mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala

```diff
@@ -205,15 +233,21 @@ class LinearSVC @Since("2.2.0") (
       val costFun = new LinearSVCCostFun(instances, $(fitIntercept),
         $(standardization), bcFeaturesStd, regParamL2, $(aggregationDepth))
 
-      def regParamL1Fun = (index: Int) => 0D
-      val optimizer = new BreezeOWLQN[Int, BDV[Double]]($(maxIter), 10, regParamL1Fun, $(tol))
+      val optimizerAlgo = $(optimizer) match {
```

Comment:
Thanks for providing the info @yanboliang.  I'm thinking we should provide both LBFGS and OWNQN as options for L1-SVM (Standard hinge). From the several dataset I tested, LBFGS converges as good as (or even better than) OWNQN and uses part of the time. Yet due to the potential failure, we may not use LBFGS as the default optimizer(solver).

A better solution as I see is to use L2-SVM(squared hinge loss) with LBFGS (combine this with https://github.com/apache/spark/pull/17645), but the change may be too large to be included into 2.2 now.